### PR TITLE
Namespace and executable changes

### PR DESCRIPTION
**Pull request by AlexejheroYTB**

### CODE REVIEW

File: QModInstaller/QModInjector.cs [hunk 1]
```diff
@@ -32,117 +30,95 @@ public QModInjector(string dir, string managedDir = null)
...
-                var patchMethod = installer.MainModule.GetType("QModInstaller.QModPatcher").Methods.FirstOrDefault(x => x.Name == "Patch");
+                AssemblyDefinition installer = AssemblyDefinition.ReadAssembly(installerFilename);
+                MethodDefinition patchMethod = installer.MainModule.GetType("QModInstaller.QModPatcher").Methods.Single(x => x.Name == "Patch");
 
-                bool patched = false;
+                TypeDefinition type = game.MainModule.GetType("TankCamera");
+                MethodDefinition method = type.Methods.Single(x => x.Name == "Awake");
 
                 foreach (var instruction in method.Body.Instructions)
                 {
```
Comment (PrimeSonic) [inline QModInstaller/QModInjector.cs]: Double check the namespace/class name of the method call in the IL instruction check below

Comment (AlexejheroYTB) [reply QModInstaller/QModInjector.cs]: At first, I was thinking about removing the old patch (if there was one), then adding the new patch call. I changed my mind and I added the old namespace back with a dummy method which calls the original one. I will move it to another file

Comment (AlexejheroYTB) [reply QModInstaller/QModInjector.cs]: If that's not what you are talking about, please clarify

File: QModInstaller/QModInjector.cs [hunk 2]
```diff
@@ -32,117 +30,95 @@ public QModInjector(string dir, string managedDir = null)
...
-                // inject
                 method.Body.GetILProcessor().InsertBefore(method.Body.Instructions[0], Instruction.Create(OpCodes.Call, method.Module.Import(patchMethod)));
 
-                // save changes under original filename
                 game.Write(mainFilename);
 
-                if (!Directory.Exists(terraTechDirectory + @"\QMods"))
-                    Directory.CreateDirectory(terraTechDirectory + @"\QMods");
+                if (!Directory.Exists(gameDirectory + @"\QMods"))
+                    Directory.CreateDirectory(gameDirectory + @"\QMods");
```
[the remark this answers is not in the document]
Comment (AlexejheroYTB) [reply QModInstaller/QModInjector.cs]: :+1:

File: QModInstaller/QModInjector.cs [hunk 3]
```diff
@@ -1,28 +1,26 @@
...
-        private string managedDirectory;
-        private string installerFilename = @"QModInstaller.dll";
-        private string mainFilename = @"\Assembly-CSharp.dll";
-        private string backupFilename = @"\Assembly-CSharp.qoriginal.dll";
-
+        public string gameDirectory;
+        public string managedDirectory;
+        public string installerFilename = @"QModInstaller.dll";
+        public string mainFilename = @"\Assembly-CSharp.dll";
+        public string backupFilename = @"\Assembly-CSharp.qoriginal.dll";
```
[the remark this answers is not in the document]
Comment (AlexejheroYTB) [reply QModInstaller/QModInjector.cs]: :+1:

File: QModInstaller/QModInjector.cs (hunk 1 above)
Comment (PrimeSonic) [reply QModInstaller/QModInjector.cs]: It's mainly just to make sure the correct method is there after all these other changes.  
But if you have the opportunity to simplify things even further, by not needing to hold onto the old method at all, I would try it out and test it.

**Comment by PrimeSonic** [comment]
Given my limited knowledge of how this all works, it looks fine to me.  
The potential issues I pointed out have been addressed.

**Comment by Aceba1** [comment]
Many `WriteLine` commands can be replaced with just one, but that's not important.

Also, you should be able to remove the `@` from paths without consequence:
Strings which start with `@` defines the string as a _verbatim string literal_. Which means that the string is taken as-is. [Source](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/verbatim).

I tried installing this to the latest update of TerraTech and noticed that there was no mod list. Was this caused by an exception?

**Comment by Aceba1** [comment]
i didn't notice that I pushed that commit with a WriteLine change...

**Comment by AlexejheroYTB** [comment]
@aceba1 Did it not log at all?

**Comment by AlexejheroYTB** [comment]
Check the output log for any exceptions, they should appear there

**Comment by Aceba1** [comment]
The mods loaded, but there was no thrown exception other than the mod (due to a namespace change)

**Comment by Aceba1** [comment]
I tested removing the mod that threw an exception, and the list printed properly.

**Comment by AlexejheroYTB** [comment]
Why should a mod react to a namespace change?
Probably it was using the mod helper, and it got removed because its not on this branch

**Comment by AlexejheroYTB** [comment]
There is another place where this needs to be modified

**Comment by AlexejheroYTB** [comment]
Same method, but in the other block (If/else)

**Comment by AlexejheroYTB** [comment]
Last call before merging!

**Pull request merged**